durable-storage/src/persistence_layer.rs

```diff
@@ -149,23 +182,68 @@ impl PersistenceLayer {
         })
     }
 
+    /// Whenever a mutable operation is performed, we need to ensure that the persistence layer is
+    /// in [`Mode::Temporary`] mode.
+    fn make_temporary(&mut self) -> Result<(), Error> {
+        match &self.mode {
+            Mode::Temporary { .. } => (),
+            Mode::FromCommit { repo } => {
+                *self = self.try_clone(repo)?;
+            }
+        };
+
+        Ok(())
+    }
+
+    /// Checks out a specific commit in the repository from the given `repo`
+    pub fn checkout(repo: &'a DirectoryManager, id: &CommitId) -> Result<Self, Error> {
+        let db_path = repo.commit_dir(id);
+
+        // We assume the commit is not found if the folder does not exist.
+        if !Path::exists(&db_path) {
+            return Err(Error::CommitNotFound);
+        };
+
+        let db = rocksdb::DB::open(&rocksdb::Options::default(), &db_path)?;
```

Comment:
Should you use the `rocksdb_options()` here instead of the default ones?


```suggestion
        let db = rocksdb::DB::open(&rocksdb_options(), &db_path)?;
```